Firestore/core/src/remote/remote_store.cc

```diff
@@ -77,6 +79,15 @@ void RemoteStore::Start() {
   // For now, all setup is handled by `EnableNetwork`. We might expand on this
   // in the future.
   EnableNetwork();
+
+  connectivity_monitor_->AddCallback(
+      [this](ConnectivityMonitor::NetworkStatus /*ignored*/) {
+        if (CanUseNetwork()) {
+          LOG_DEBUG("RemoteStore %s restarting streams as connectivity changed",
+                    this);
+          RestartNetwork();
```

Comment:
This doesn't create any undesirable interactions with the existing logic on connectivity change, right? (I don't think it should, just checking)

Reply:
The existing logic invalidates the GRPC channel, which speeds up reconnects already and is the foundation this PR is based on.